### PR TITLE
Fixed #33033 -- Prevented models.DecimalField from accepting NaN values.

### PR DESCRIPTION
**Pull request by chinmoy12c**
Ticket reference: https://code.djangoproject.com/ticket/33033

### CODE REVIEW

File: django/db/models/fields/__init__.py [hunk 1]
```diff
@@ -1539,6 +1540,11 @@ def to_python(self, value):
         if value is None:
             return value
         if isinstance(value, float):
+            if (isnan(value)):
```
Comment (timgraham) [inline django/db/models/fields/__init__.py:1543]: outer parentheses can be removed

File: django/db/models/fields/__init__.py [hunk 2]
```diff
@@ -1539,6 +1540,11 @@ def to_python(self, value):
...
+                raise exceptions.ValidationError(
+                    'NaN value is not supported',
```
Comment (timgraham) [inline django/db/models/fields/__init__.py:1545]: This string should be in `default_error_messages` and be marked as translatable with the underscore: `_('...')`, although I think it's probably okay to reuse the `'invalid'` message... the message "nan value must be a decimal number." seems to read okay.

File: tests/model_fields/test_decimalfield.py [hunk 3]
```diff
@@ -100,3 +101,12 @@ def test_roundtrip_with_trailing_zeros(self):
         obj = Foo.objects.create(a='bar', d=Decimal('8.320'))
         obj.refresh_from_db()
         self.assertEqual(obj.d.compare_total(Decimal('8.320')), Decimal('0'))
+
+    def test_nan_value(self):
+        floatNaN = BigD(d=float('nan'))
```
Comment (timgraham) [inline tests/model_fields/test_decimalfield.py:106]: I think names like `float_nan` would be more consistent with our coding style.

**Comment by felixxm** [comment]
@chinmoy12c Thanks :+1:

I rebased and pushed small edits.

**Pull request merged**